src/simplewallet/simplewallet.cpp

```diff
@@ -3854,6 +3856,75 @@ bool simple_wallet::check_tx_proof(const std::vector<std::string> &args)
   return check_tx_key_helper(txid, address, derivation);
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::get_spend_proof(const std::vector<std::string> &args)
+{
+  if(args.size() != 1) {
+    fail_msg_writer() << tr("usage: get_spend_proof <txid>");
+    return true;
+  }
+
+  if (m_wallet->watch_only())
+  {
+    fail_msg_writer() << tr("wallet is watch-only and cannot generate the proof");
+    return true;
+  }
+
+  cryptonote::blobdata txid_data;
+  if(!epee::string_tools::parse_hexstr_to_binbuff(args[0], txid_data) || txid_data.size() != sizeof(crypto::hash))
+  {
+    fail_msg_writer() << tr("failed to parse txid");
+    return true;
+  }
+  crypto::hash txid = *reinterpret_cast<const crypto::hash*>(txid_data.data());
```

Comment:
This `reinterpret_cast` is technically UB, but _should_ work. The call `hex_to_pod` does the same thing! Anyway, I just wanted to point out that you can drop a lot of this code by using `hex_to_pod`.

Reply:
Ah, good to know about a simpler alternative! Thanks.